bee-ternary/src/bigint/binary_representation.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+//! Binary representation of big integers.
+
+use crate::bigint::sealed::Sealed;
+
+/// The number of bits in an U384/I384.
+pub const BINARY_LEN: usize = 384;
+
+/// Binary representation of a big integer.
+pub trait BinaryRepresentation: Sealed + Clone {
+    /// Inner representation type of the big integer.
+    type T;
+
+    /// Iterates over a slice of the inner representation type of the big integer.
+    fn iter(&self) -> std::slice::Iter<'_, Self::T>;
+}
+
+/// The number of u8s in an U384/I384.
+pub const BINARY_LEN_IN_U8: usize = BINARY_LEN / 8;
+
+/// The inner representation of an U384/I384 using 48 u8s.
+pub type U8Repr = [u8; BINARY_LEN_IN_U8];
+
+impl Sealed for U8Repr {}
+
+impl BinaryRepresentation for U8Repr {
```

Comment:
It is actually a binary representation. The fact the the trait currently only provides `iter` doesn't make it an iterator. See it more as a marker than something else. Other types have bounds on that stuff.

Reply:
oh, I was probably not clear enough. I am not saying it is an iterator despite my question at the end. I said, that is more like an `AsIterator`. Is the trait supposed to be expanded? I am just basing my argument on what I currently see, what the trait enforces, and that is just some way to iterator over a slice of the inner type, hence my confusion about the name. I don't see anything in there that relates to binary representations.

Reply:
meaning I could take that `trait` and use it a completely different context that has nothing to do with binary/ternary representations. If you think this is the proper name, then go with it. I don't mind that much.

Reply:
No you couldn't necessarily use in in a completely different context. Look at the iteration, it's not a classic iter. It iterates depending on the inner binary representation. It can iterate in `u8` or `u32`. And the trait itself is necessary because it is a bound for other things.

Reply:
alright then :slightly_smiling_face: 
Thanks for the discussion. We can close this I think.